io.openems.edge.battery.api/src/io/openems/edge/battery/statemachine/ErrorHandler.java

```diff
@@ -0,0 +1,34 @@
+package io.openems.edge.battery.statemachine;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.edge.battery.statemachine.StateMachine.State;
+import io.openems.edge.common.statemachine.StateHandler;
+
+public class ErrorHandler extends StateHandler<State, Context> {
+
+	private Instant entryAt = Instant.MIN;
+
+	@Override
+	protected void onEntry(Context context) throws OpenemsNamedException {
+		this.entryAt = Instant.now();
+	}
+
+	@Override
+	protected void onExit(Context context) throws OpenemsNamedException {
+		var battery = context.getParent();
+		battery._setMaxStartAttempts(false);
+		battery._setMaxStopAttempts(false);
+	}
+
+	@Override
+	public State runAndGetNextState(Context context) {
+		if (Duration.between(this.entryAt, Instant.now()).getSeconds() > 120) {
```

Comment:
you could make the `120` as a constant maybe `SECONDS_UNTIL_RETRY` to make it more readable